### PR TITLE
doc: upgrade docs/life_of_a_godwoken_transaction.md to v1.1

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: docs/life_of_a_godwoken_transaction.md [hunk 1]
```diff
@@ -7,48 +7,51 @@ Godwoken is a layer 2 rollup framework for [Nervos CKB](https://github.com/nervo
...
+Fundamentally, Godwoken is a layer 2 solution for layer 1 Nervos CKB blockchain. [Here](https://pudge.explorer.nervos.org/address/ckt1qpuljza4azfdsrwjzdpea6442yfqadqhv7yzfu5zknlmtusm45hpuqgqcfn63wfumts4lvrryhc35u43q3aaf5euqq6zeas9) shows one deployment of Godwoken on CKB's testnet. A new layer 2 Godwoken block, packed in a layer 1 CKB transaction, is issued to CKB's network periodically. The current state of the layer 2 Godwoken blockchain is committed in the cell data part of a single CKB cell. [This layer 1 CKB transaction](https://pudge.explorer.nervos.org/transaction/0x9fd03b09d27326aa54dc53e11d7f7c01f27e570bdc645233b91664323cac0568) serves as an example of the layer 2 Godwoken block. The chain state, as of this layer 2 block, is stored in the cell data part of the first output cell in this transaction:
```
Comment (blckngm) [inline docs/life_of_a_godwoken_transaction.md]:
```suggestion
Fundamentally, Godwoken is a layer 2 solution for layer 1 Nervos CKB blockchain. [Here](https://pudge.explorer.nervos.org/address/ckt1qpuljza4azfdsrwjzdpea6442yfqadqhv7yzfu5zknlmtusm45hpuqgqcfn63wfumts4lvrryhc35u43q3aaf5euqq6zeas9) shows one deployment of Godwoken on CKB's testnet. New layer 2 Godwoken blocks are submitted to layer 1 CKB in **block submission transactions**. The current state of the layer 2 Godwoken blockchain is committed in the cell data part of the CKB cell generated by the block submission transaction. [This transaction](https://pudge.explorer.nervos.org/transaction/0x9fd03b09d27326aa54dc53e11d7f7c01f27e570bdc645233b91664323cac0568) is an example of a block submission transaction. The chain state, as of this layer 2 block, is:
```

File: docs/life_of_a_godwoken_transaction.md [hunk 2]
```diff
@@ -7,48 +7,51 @@ Godwoken is a layer 2 rollup framework for [Nervos CKB](https://github.com/nervo
...
     // 0: running, 1: halting
     status: byte,
+    version: byte,
 }
 ```
 
-Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L18) for more definitions.
+Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol) for more definitions.
 
 The actual layer 2 Godwoken block, is also contained in the layer 1 CKB transaction. The first witness of the above mentioned transaction, contains the following data:
```
Comment (blckngm) [inline docs/life_of_a_godwoken_transaction.md]:
```suggestion
The actual layer 2 Godwoken block, is also contained in the block submission transaction. The first witness of the above mentioned transaction, contains the following data:
```

File: docs/life_of_a_godwoken_transaction.md [hunk 3]
```diff
@@ -85,55 +88,35 @@ Note this is slightly different from a typical optimistic rollup: here a transac
...
 
 ## Deposit
 
-To use Godwoken, one must first deposit some tokens(either CKB or sUDTs) from layer 1 to layer 2. This is named as a **deposit** action. A deposit action is represented as a layer 1 CKB transaction, it must create a special output cell named a **deposit cell**. [Here](https://explorer.nervos.org/aggron/transaction/0x09334d0ab2bf84714f0e51715bb7251a4887aa3066ef5ad0770bd463297c1194) is an example of a deposit action. This transaction deposits 10000 CKB to Godwoken. What is relavant here, is the lock script of output cell #0:
+To use Godwoken, one must first deposit some tokens(either CKB or sUDTs) from layer 1 to layer 2. This is named as a **deposit** action. A deposit action is represented as a layer 1 CKB transaction, it must create a special output cell named a **deposit cell**. [Here](https://pudge.explorer.nervos.org/transaction/0xb3aceeb1d84ea5b24e4b8fe339fc6e7814f1e58ebc99237068f22a6924501a1e) is an example of a deposit action. This transaction deposits 10000 CKB to Godwoken. What is relavant here, is the lock script of output cell #0:
```
Comment (blckngm) [inline docs/life_of_a_godwoken_transaction.md:101]: Is it tokens or assets (used in the deposit and withdrawal doc)? Shall we consolidate on one term?

**Review by blckngm — COMMENTED** [review]
Added a new concept: block submission transactions. What do you think?

File: docs/life_of_a_godwoken_transaction.md (hunk 3 above)
Comment (jjyr) [reply docs/life_of_a_godwoken_transaction.md:101]: Tokens and assets are both used wildly in blockchain products, I think it is ok to mix use the two words.

**Review by blckngm — DISMISSED** [review]

File: docs/life_of_a_godwoken_transaction.md [hunk 4]
```diff
@@ -397,27 +393,14 @@ Here is the JSON representation of a `WithdrawalRequest`:
 {
   "raw": {
     "nonce": "0x3",
+    "chain_id": "0x0",
```
Comment (Flouse) [inline docs/life_of_a_godwoken_transaction.md]: consider filling this value with the CHAIN_ID of Godwoken testnet_v1?

**Review by Flouse — DISMISSED** [review]

**Review by Flouse — APPROVED** [review]

**Review by blckngm — APPROVED** [review]

File: docs/life_of_a_godwoken_transaction.md [hunk 5]
```diff
@@ -7,48 +7,51 @@ Godwoken is a layer 2 rollup framework for [Nervos CKB](https://github.com/nervo
...
+Fundamentally, Godwoken is a layer 2 solution for layer 1 Nervos CKB blockchain. [Here](https://pudge.explorer.nervos.org/address/ckt1qpuljza4azfdsrwjzdpea6442yfqadqhv7yzfu5zknlmtusm45hpuqgqcfn63wfumts4lvrryhc35u43q3aaf5euqq6zeas9) shows one deployment of Godwoken on CKB's testnet. New layer 2 Godwoken blocks are submitted to layer 1 CKB in **block submission transactions**. The current state of the layer 2 Godwoken blockchain is committed in the cell data part of the CKB cell generated by the block submission transaction. [This transaction](https://pudge.explorer.nervos.org/transaction/0x9fd03b09d27326aa54dc53e11d7f7c01f27e570bdc645233b91664323cac0568) is an example of a block submission transaction. The chain state, as of this layer 2 block, is:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:10]: Godwoken is fundamentally a layer 2 solution for layer 1, the Nervos CKB blockchain. [Here](https://pudge.explorer.nervos.org/address/ckt1qpuljza4azfdsrwjzdpea6442yfqadqhv7yzfu5zknlmtusm45hpuqgqcfn63wfumts4lvrryhc35u43q3aaf5euqq6zeas9) is an example of a Godwoken deployment on the CKB testnet. New layer 2 blocks are submitted to layer 1 in **block submission transactions**. The current state of the Godwoken blockchain is committed in the cell data of the CKB cells generated by the block submission transactions. [This transaction](https://pudge.explorer.nervos.org/transaction/0x9fd03b09d27326aa54dc53e11d7f7c01f27e570bdc645233b91664323cac0568) represents an example of a block submission transaction. The chain state of this layer 2 block is as follows:

File: docs/life_of_a_godwoken_transaction.md [hunk 6]
```diff
@@ -7,48 +7,51 @@ Godwoken is a layer 2 rollup framework for [Nervos CKB](https://github.com/nervo
...
     status: byte,
+    version: byte,
 }
 ```
 
-Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L18) for more definitions.
+Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol) for more definitions.
 
-The actual layer 2 Godwoken block, is also contained in the layer 1 CKB transaction. The first witness of the above mentioned transaction, contains the following data:
+The actual layer 2 Godwoken block, is also contained in the block submission transaction. The first witness of the above mentioned transaction, contains the following data:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:35]: The block submission transaction also contains the actual layer 2 block. The first witness of the transaction contains the following data:

File: docs/life_of_a_godwoken_transaction.md [hunk 7]
```diff
@@ -7,48 +7,51 @@ Godwoken is a layer 2 rollup framework for [Nervos CKB](https://github.com/nervo
...
 ```
 
-This is actually a `WitnessArgs` data structure described [here](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/blockchain.mol#L105), and serialized in [molecule](https://github.com/nervosnetwork/molecule)'s format. In the `output_type` field of WitnessArgs, the actual `L2Block` data structure is stored, also serialized in molecule's format:
+This is actually a `WitnessArgs` data structure described [here](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/blockchain.mol#L116), and serialized in [molecule](https://github.com/nervosnetwork/molecule)'s format. In the `output_type` field of WitnessArgs, the actual `L2Block` data structure is stored, also serialized in molecule's format:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:41]: The [WitnessArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/blockchain.mol#L116) data structure is serialized in the molecule format. In the `output_type` field of `WitnessArgs`, the actual `L2Block` data structure is stored and serialized in the molecule format:

File: docs/life_of_a_godwoken_transaction.md [hunk 8]
```diff
@@ -66,11 +69,11 @@ Godwoken provides an account model to Nervos CKB:
...
+One might wonder where all the data are stored given constant storage requirement in the above transactions. Godwoken leverages [Sparse Merkle Tree](https://medium.com/@kelvinfichter/whats-a-sparse-merkle-tree-acda70aeb837) to build the storage space. Only the root hash of the Sparse Merkle Tree(SMT) is kept in the rollup cell. State validator validates that the sparse merkle tree is correctly updated in each layer 2 block. We actually took a step further here, and build an optimized SMT, which can save a lot of space and calculation for CKB's use case. Please refer to [here](https://github.com/nervosnetwork/sparse-merkle-tree) for more details.
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:72]: Considering the constant storage requirements of the above transactions, you may wonder where all the data is stored. Godwoken leverages [Sparse Merkle Tree](https://medium.com/@kelvinfichter/whats-a-sparse-merkle-tree-acda70aeb837) to build the storage space. The rollup cell only contains the root hash of the Sparse Merkle Tree (SMT). The state validator verifies that the sparse merkle tree is correctly updated in each layer 2 block. We have actually gone one step further here, and built an optimized SMT, which will save CKB a great deal of space and calculation. For more details, see [here](https://github.com/nervosnetwork/sparse-merkle-tree).

File: docs/life_of_a_godwoken_transaction.md [hunk 9]
```diff
@@ -66,11 +69,11 @@ Godwoken provides an account model to Nervos CKB:
...
 
 ## Optimistic Rollup
 
-Godwoken by design, only defines a generic rollup framework. Both optimistic rollup and zero knowledge rollup can be supported by Godwoken. However, in the current version, we are primarily focusing on optimistic rollup. The following workflow is used by Godwoken:
+The following workflow is used by Godwoken:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:76]: Godwoken uses the following workflow:

File: docs/life_of_a_godwoken_transaction.md [hunk 10]
```diff
@@ -85,55 +88,35 @@ Note this is slightly different from a typical optimistic rollup: here a transac
 
 In this design, an aggregator bears liquidity costs for staking CKBs, as well as layer 1 transaction fees for the layer 1 transactions containing layer 2 block. In exchange, the aggregator can charge layer 2 transaction fees from layer 2 users. As explained below, layer 2 transaction fees can be paid either in CKB or in any kind of [sUDT](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0025-simple-udt/0025-simple-udt.md). In fact, Godwoken treat CKB and sUDTs as exactly the same thing in layer 2.
 
-## PoA Deployments
+## Multiple block producers
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:91]: Multiple Block Producers

File: docs/life_of_a_godwoken_transaction.md [hunk 11]
```diff
@@ -85,55 +88,35 @@ Note this is slightly different from a typical optimistic rollup: here a transac
...
-* The aggregators can also start a vote to update the contents in the PoASetup cell. For example, when an aggregator behaves badly, either by not issuing blocks, or issuing too many invalid blocks, the other aggregators can start a vote to upgrade the PoASetup cell, hence removing the misbehaved aggregator. Another example is that in the event of congested network, aggregators can vote to alter the round intervals for better performance.
+Godwoken is designed based on the assumption that anyone shall be able to propose layer2 blocks. But in the current deployment, Godwoken has only one block producer to sequence transactions and propose new blocks. To support multiple block producers (or sequencers) is still an open question for different rollups. In the future, we may introduce a consensus through upgrading to support multiple block producer coordination.
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:93]: Godwoken was designed assuming anyone can propose layer 2 blocks. However, in the current deployment, Godwoken only has one block producer to sequence transactions and propose new blocks. Support for multiple block producers (or sequencers) is still an open question for different rollups. In the future, we may introduce a consensus through an upgrade to support multiple block producers.

File: docs/life_of_a_godwoken_transaction.md (hunk 3 above)
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:101]: In order to use Godwoken, you will be required to deposit some tokens, CKB or sUDT, from layer 1 to layer 2. This is known as a **deposit** action. A deposit action is represented as a layer 1 transaction. It must create a special output cell called a **deposit cell**. [Here](https://pudge.explorer.nervos.org/transaction/0xb3aceeb1d84ea5b24e4b8fe339fc6e7814f1e58ebc99237068f22a6924501a1e) is an example of a deposit action. This transaction deposits 10,000 CKB to Godwoken. The lock script of the output cell #0 is as follows:

File: docs/life_of_a_godwoken_transaction.md [hunk 12]
```diff
@@ -85,55 +88,35 @@ Note this is slightly different from a typical optimistic rollup: here a transac
...
     "hash_type": "type"
 }
 ```
 
-`code_hash` and `hash_type` is pre-determined by each Godwoken deployment. `args` contains 2 parts: the first 32 bytes of `args`, contain the **rollup type hash** of the current Godwoken deployment(remember we mentioned earlier, that rollup type hash works like a chain ID?). The second part, is the [DepositLockArgs](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L121) data structure, serialized in molecule format:
+`code_hash` and `hash_type` is pre-determined by each Godwoken deployment. `args` contains 2 parts: the first 32 bytes of `args`, contain the **rollup type hash** of the current Godwoken deployment(remember we mentioned earlier, that rollup type hash works like a chain ID?). The second part, is the [DepositLockArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L177) data structure, serialized in molecule format:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:111]: `code_hash` and `hash_type` are pre-determined by each Godwoken deployment. `args` contains two parts: **rollup type hash** and **DepositLockArgs**. The first 32 bytes of `args` contain the **rollup type hash** of the current Godwoken deployment, which functions as a chain ID. The [DepositLockArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L177) data structure is serialized in the molecule format:

File: docs/life_of_a_godwoken_transaction.md [hunk 13]
```diff
@@ -144,38 +127,38 @@ This data structure contains 2 parts of information:
...
     "hash_type": "type"
 }
 ```
 
-Like deposit cells, custodian cells have pre-determined `code_hash` and `hash_type` based on Godwoken deployments. The first 32 bytes in `args` contain the rollup type hash as well. What's different here, is that [CustodianLockArgs](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L131) is used instead to fill the remaining part of `args`:
+Like deposit cells, custodian cells have pre-determined `code_hash` and `hash_type` based on Godwoken deployments. The first 32 bytes in `args` contain the rollup type hash as well. What's different here, is that [CustodianLockArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L188) is used instead to fill the remaining part of `args`:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:140]: Like deposit cells, custodian cells have pre-determined `code_hash` and `hash_type` based on Godwoken deployments. The first 32 bytes in `args` contain the rollup type hash as well. What is different here is that [CustodianLockArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L188) is used to fill the remaining part of `args`:

File: docs/life_of_a_godwoken_transaction.md [hunk 14]
```diff
@@ -184,11 +167,11 @@ table RawL2Transaction {
 
 table L2Transaction {
     raw: RawL2Transaction,
-    signature: Signature,
+    signature: Bytes,
 }
 ```
 
-Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L27-L37) for a more complete definition and related types.
+Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L69-L81) for a more complete definition and related types.
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:174]: The full definition and related types can be found in [this file](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L69-L81).

File: docs/life_of_a_godwoken_transaction.md [hunk 15]
```diff
@@ -206,13 +190,15 @@ While all operations on layer 2 Godwoken can be represented as a `L2Transaction`
 }
 ```
 
-This is in fact the JSON representation of the above `L2Transaction` data structure. `args` contains a variable length, free formatted transaction argument data that is interpreted depending on the values of `to_id`. For this particular example, `args` contains [SUDTTransfer](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L211-L215) data structure in molecule serialization format:
+This is in fact the JSON representation of the above `L2Transaction` data structure. `args` contains a variable length, free formatted transaction argument data that is interpreted depending on the values of `to_id`. For this particular example, `args` contains [SUDTTransfer](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L260) data structure in molecule serialization format:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:193]: This is the JSON representation of the `L2Transaction` data structure. `args` contains a variable length, free formatted transaction argument data that is interpreted depending on the values of `to_id`. In this particular example, `args` contains the [SUDTTransfer](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L260) data structure in the molecule serialization format:

File: docs/life_of_a_godwoken_transaction.md [hunk 16]
```diff
@@ -222,14 +208,14 @@ A JSON representation for this data structure is:
...
-      "to": "0x4",
+      "to_address": "0x0200000014000000bb1d13450cfa630728d0390c99957c6948bf7d19",
       "amount": "0x190",
       "fee": "0x64"
   }
 }
 ```
 
-In this example, we are transferring 400 shannons(denoted by `to_id` 0x1) from account 0x2(denoted by `from_id`) to account 0x4(denoted by `to` in `SUDTTransfer`), we would also love to pay for 100 shannons as layer 2 transaction fees. The next 2 sections shall explain the details for layer 2 transaction.
+In this example, we are transferring 400 shannons(denoted by `to_id` 0x1) from account 0x2(denoted by `from_id`) to account 0x0200000014000000bb1d13450cfa630728d0390c99957c6948bf7d19(denoted by `to_address` in `SUDTTransfer`), we would also love to pay for 100 shannons as layer 2 transaction fees. The next 2 sections shall explain the details for layer 2 transaction.
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:218]: In this example, we are transferring 400 shannons (denoted by `to_id` 0x1) from account 0x2 (denoted by `from_id`) to account 0x0200000014000000bb1d13450cfa630728d0390c99957c6948bf7d19 (denoted by `to_address` in `SUDTTransfer`). In addition, we would like to pay 100 shannons as layer 2 transaction fees. The following two sections will discuss layer 2 transactions in more detail.

File: docs/life_of_a_godwoken_transaction.md [hunk 17]
```diff
@@ -263,7 +249,7 @@ Now we can derive the signature validation rule for a layer 2 Godwoken transacti
...
+There is one quirk to the above flow here: recall that current Godwoken uses optimistic rollup design. Due to the "optimistic" nature, the layer 2 lock script is not typically executed on chain. It is only executed in the case that a challenger starts a challenge request on chain, and an aggregator proves the validity of the layer 2 transaction via a cancel challenge request. Hence the way to build a layer 2 lock script, is also slightly different here. An example for such a script, can be found [here](https://github.com/nervosnetwork/godwoken-scripts/blob/master/contracts/eth-account-lock/src/entry.rs#L24).
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:252]: The above flow has one quirk: the current version of Godwoken uses optimistic rollup design. Due to the "optimistic" nature, the layer 2 lock script is not typically executed on chain. It is only executed when a challenger initiates a challenge request on chain, and an aggregator validates the validity of the layer 2 transaction through a cancel challenge request. Hence the way to build a layer 2 lock script is also slightly different. An example of such a script can be found [here](https://github.com/nervosnetwork/godwoken-scripts/blob/master/contracts/eth-account-lock/src/entry.rs#L24).

File: docs/life_of_a_godwoken_transaction.md [hunk 18]
```diff
@@ -276,7 +262,7 @@ Godwoken actually has 2 kinds of accounts:
...
+While a typical user account is a balance owned by a user, contract account, on the other hand, provides a storage for an on-chain smart contract. Similar to the way that a user account is created from a general lock script with a user's public key hash, a smart contract for a contract account is created from a **backend** with some special script args. The smart contract is also represented using the unified [Script](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/blockchain.mol#L32) data structure in the Nervos ecosystem: the combination of `code_hash` and `hash_type` in a script identifies the backend, while `args` provide contract account specific arguments.
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:265]: While a typical user account is a balance owned by a user, a contract account, on the other hand, provides storage for an on-chain smart contract. As a user account is generated from a general lock script with a user's public key hash, a smart contract for a contract account is generated from a **backend** with some special script args. In the Nervos ecosystem, smart contracts are also represented using the unified [Script](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/blockchain.mol#L32) data structure: the combination of `code_hash` and `hash_type` in a script identifies the backend, while `args` provides specific contract account arguments.

File: docs/life_of_a_godwoken_transaction.md [hunk 19]
```diff
@@ -367,25 +356,32 @@ If there is one particular account lock or backend you like, please do not hesis
...
 
 struct WithdrawalRequest {
     raw: RawWithdrawalRequest,
-    signature: Signature,
+    signature: Bytes,
 }
 ```
 
-Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L71-L94) for a more complete definition and related types.
+Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L136-L160) for a more complete definition and related types.
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:384]: The full definition and related types can be found in [this file](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L136-L160).

File: docs/life_of_a_godwoken_transaction.md [hunk 20]
```diff
@@ -327,7 +314,7 @@ Here is the JSON representation for a layer 2 transaction, which invokes MetaCon
 }
 ```
 
-The `args` part in this transaction, contains a [MetaContractArgs](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/crates/types/schemas/godwoken.mol#L192-L194) data structure serialized in molecule format. A JSON representation for the `args`, is as follows:
+The `args` part in this transaction, contains a [MetaContractArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L232) data structure serialized in molecule format. A JSON representation for the `args`, is as follows:
```
Comment (xying21) [inline docs/life_of_a_godwoken_transaction.md:317]: The `args` part of this transaction contains a [MetaContractArgs](https://github.com/nervosnetwork/godwoken/blob/v1.1.0-beta/crates/types/schemas/godwoken.mol#L232) data structure serialized in the molecule format. A JSON representation of the `args` is as follows:

**Review by xying21 — CHANGES_REQUESTED** [review]
Finished the review

File: docs/life_of_a_godwoken_transaction.md (hunk 18 above)
Comment (ashleewang888) [reply docs/life_of_a_godwoken_transaction.md:265]: A typical user account is a balance owned by the user, while a contract account provides storage for the on-chain smart contracts. The user account is generated from a general lock script with a user's public key hash, while the smart contract for the contract account is generated from a **backend** with some special script args.

**Pull request merged**